kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaTopicManager.java

```diff
@@ -379,14 +382,21 @@ public void deReference(String topicName) {
     }
 
     public CompletableFuture<Consumer> getGroupConsumers(String groupId, TopicPartition kafkaPartition) {
-        // make sure internal consumer existed
-        CompletableFuture<Consumer> consumerFuture = new CompletableFuture<>();
-        if (groupId == null || groupId.isEmpty() || !requestHandler.getGroupCoordinator()
-                .getOffsetAcker().getConsumer(groupId, kafkaPartition).isDone()) {
-            log.warn("not get consumer for group {} this time", groupId);
-            consumerFuture.complete(null);
-            return consumerFuture;
+        if (StringUtils.isEmpty(groupId)) {
+            log.warn("Try to get group consumers with an empty group id");
+            return CompletableFuture.completedFuture(null);
         }
+
+        // The future of the offset consumer should be created before in `GroupCoordinator#handleSyncGroup`
+        final OffsetAcker offsetAcker = requestHandler.getGroupCoordinator().getOffsetAcker();
+        final CompletableFuture<org.apache.pulsar.client.api.Consumer<byte[]>> offsetConsumerFuture =
+                offsetAcker.getConsumer(groupId, kafkaPartition);
+        if (offsetConsumerFuture == null) {
+            log.warn("No offset consumer for [group={}] [topic={}]", groupId, kafkaPartition);
```

Comment:
It may be better to change this log level to debug, I found a lot of this warning in the test environment and will check again.

Reply:
I agree. there will be a few of this log if the consumer is not ready.